### ISSUE TITLE
Overfitting Prevention

### ISSUE THREAD

**Issue by Enzogall** [MathEpitech2025/my_torch#10]
Implement overfitting prevention techniques:

* Train/validation dataset split
* Early stopping mechanism
* Monitor validation loss during training
* Regularization techniques (L1/L2)
